modules/lang-painless/src/main/java/org/elasticsearch/painless/Debug.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.painless;
+
+/**
+ * Utility methods for debugging painless scripts that are accessible to painless scripts.
+ */
+public class Debug {
+    private Debug() {}
+
+    /**
+     * Throw an {@link Error} that "explains" an object.
+     */
+    public static void explain(Object objectToExplain) throws PainlessExplainError {
+        String className = objectToExplain == null ? "null" : objectToExplain.getClass().getName();
+        String message = "getClass=[" + className + "] toString=[" + objectToExplain + "]";
+
+        throw new PainlessExplainError(message, objectToExplain);
+    }
+
+    /**
+     * Thrown by {@link Debug#explain(Object)} to explain an object. Subclass of {@linkplain Error} so it cannot be caught by painless
+     * scripts.
+     */
+    public static class PainlessExplainError extends Error {
+        private final Object objectToExplain;
+
+        public PainlessExplainError(String message, Object objectToExplain) {
+            super(message);
+            this.objectToExplain = objectToExplain;
+        }
+
+        Object getObjectToExplain() {
```

Comment:
if you make this structured ie. a map then you can use `addHeader` on ScritpException to make rendering nice?